rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,270 @@
+
+              use: [
+                {
+                  // PostCSS is a general-purpose CSS transformer; however, we prefer to avoid custom CSS syntaxes
+                  // and only use the standard SASS syntax.  Thus postcss-loader is used here only to apply the popular
+                  // "autoprefixer" plugin improves browser compatibility by generating vendor prefixes.
+                  // https://www.npmjs.com/package/postcss-loader
+                  loader: require.resolve('postcss-loader'),
+                  options: {
+                    postcssOptions: {
+                      plugins: [
+                        // https://www.npmjs.com/package/autoprefixer
+                        autoprefixer
+                      ]
+                    },
+
+                    sourceMap: !production
+                  }
+                },
+                {
+                  // Compiles SASS syntax into CSS
+                  // https://www.npmjs.com/package/sass-loader
+                  loader: require.resolve('sass-loader'),
+
+                  options: {
+                    implementation: sass,
+                    sassOptions: {
+                      includePaths: [path.resolve(__dirname, 'node_modules')]
+                    },
+
+                    sourceMap: !production
+                  }
+                }
+              ]
+            }
+          ]
+        },
+
+        {
+          test: /\.(jpeg|jpg|png|gif|svg|ico)$/,
+          // Allows import/require() to be used with an asset file. The file will be copied to the output folder,
+          // and the import statement will return its URL.
+          // https://www.npmjs.com/package/file-loader
+          loader: 'file-loader'
+        }
+      ]
+    },
+    devServer: {
+      host: 'localhost',
+      port: 8080
+    },
+    // See here for documentation: https://webpack.js.org/configuration/devtool
+    devtool: production ? undefined : 'eval-source-map',
+    optimization: {
+      minimize: !!production,
+      nodeEnv: production ? 'production' : 'development',
+      splitChunks: {
+        chunks: 'all'
+      },
+      minimizer: [
+        new CssMinimizerPlugin({
+          minimizerOptions: {
+            preset: ['default', { discardComments: { removeAll: !!production } }]
+          }
+        }),
+        // This magic constant indicates the minimizer defaults
+        // https://webpack.js.org/configuration/optimization/#optimizationminimizer
+        '...'
+      ]
+    },
+    plugins: [
+      // See here for documentation: https://webpack.js.org/plugins/mini-css-extract-plugin/
+      new MiniCssExtractPlugin({
+        filename: '[name].[contenthash].css'
+      }),
+
+      // See here for documentation: https://webpack.js.org/plugins/define-plugin/
+      new DefinePlugin({
+        DEBUG: !production
+      }),
+
+      process.env.REPORT ? new BundleAnalyzerPlugin({ analyzerMode: 'static' }) : undefined
```

Comment:
The idea is to generate the .html reports optionally, only if the `REPORT` environment variable is set. (Other ideas welcome.)  Currently we are using [dotenv](https://www.npmjs.com/package/dotenv) at HBO.